custom_components/geappliances/ha_compatibility/meta_erds.py

```
@@ -95,15 +95,16 @@ async def enable_or_disable(
     unique_id: str,
 ) -> None:
     """Enable or disable the entity."""
-    await hass.services.async_call(
-        DOMAIN,
-        SERVICE_ENABLE_OR_DISABLE,
-        {
-            ATTR_ENTITY_ID: entity_id,
-            ATTR_UNIQUE_ID: unique_id,
-            ATTR_ENABLED: enabled_bytes != b"\x00",
-        },
-    )
+    if entity_id:
```

When is there not an `entity_id`?

---

I think on things like this we might want to look into it further. If we're calling this function without an entity ID something is wrong. I think we should either let this throw an error or at least log an error stating we didn't get an entity ID. I feel the same for the changes made to the `set_allowables` function, just skipping stuff when we receive invalid args is going to cause us to miss bugs down the road.

---

So the reason that this is happening at least for the enable/disable is that we have erd 1 with fields allowable 1, 2, and 3. Currently the logic treats erd 1 as 1 erd but 3 separate entities. So if allowable 1 gets updated to OFF i believe it still calls the enable_disable function 3 times. But after the first call the erd has already been unsubscribed so the function returns None for the rest.